src/tools/components/ResultDownload.tsx

```diff
@@ -109,13 +110,7 @@ const ResultDownload = memo<ResultDownloadProps>(
 
     const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
       e.preventDefault();
-      const link = document.createElement('a');
-      link.href = toolsURLs(jobType).resultUrl(id, fileFormat);
-      link.target = '_blank';
-      link.setAttribute('download', '');
-      document.body.appendChild(link);
-      link.click();
-      document.body.removeChild(link);
+      downloadFileInNewTab(toolsURLs(jobType).resultUrl(id, fileFormat));
```

Comment:
Not sure why this needs to be done instead of just having the submit button be a link to the output of `toolsURLs(jobType).resultUrl(id, fileFormat)` with all the attributes as set inside that `downloadFileInNewTab` function

Reply:
When the submit button is clicked we download in a new tab as well as call `onToggleDisplay` - are you thinking have the button as a link as well as having an onClick handler to deal with `onToggleDisplay`?

Reply:
Yep. As long as you don't block the normal click event you should still be able to do both things

Reply:
Updated